src/orquestra/sdk/_base/_driver/_models.py

```diff
@@ -431,3 +442,123 @@ class Event(NamedTuple):
+    """Types of sources that can emit system logs."""
+
+    RAY_HEAD_NODE = "RAY_HEAD_NODE"
+    RAY_WORKER_NODE = "RAY_WORKER_NODE"
+    K8S_EVENT = "K8S_EVENT"
+
+
+class K8sEventLogContainerInfoStatus(pydantic.BaseModel):
+    state: str
+    reason: Optional[str]
+    message: Optional[str]
+
+
+class K8sEventLogContainerInfo(pydantic.BaseModel):
+    name: str
+    status: K8sEventLogContainerInfoStatus
+
+
+class K8sEventLogPodConditions(pydantic.BaseModel):
+    type: str
+    status: str
+    reason: Optional[str]
+    message: Optional[str]
+
+
+class K8sEventLogContent(pydantic.BaseModel):
+    name: str
+    namespace: str
+    event: str
+    phase: str
+    podConditions: Optional[List[K8sEventLogPodConditions]]
+    containerInfo: Optional[List[K8sEventLogContainerInfo]]
+
+    def __str__(self) -> str:
+        str_repr = f"{self.name} {self.event} {self.phase}"
+        if self.podConditions:
+            str_repr += "\n  Pod Conditions:"
+            pclog: K8sEventLogPodConditions
+            for pclog in self.podConditions:
+                str_repr += f"\n  -- {pclog.type} {pclog.status}"
+                if pclog.reason:
+                    str_repr += f" ({pclog.reason})"
+                if pclog.message:
+                    str_repr += f": {pclog.message} "
+        if self.containerInfo:
+            str_repr += "\n  Container Information:"
+            cilog: K8sEventLogContainerInfo
+            for cilog in self.containerInfo:
+                str_repr += f"\n  -- {cilog.name} {cilog.status.state}"
+                if cilog.status.reason:
+                    str_repr += f" ({cilog.status.reason})"
+                if cilog.status.message:
+                    str_repr += f": {cilog.status.message} "
+        return str_repr
+
+
+class K8sEventLog(pydantic.BaseModel):
+    """A system-level log line produced by a K8S event."""
+
+    tag: str
+
+    log: K8sEventLogContent
+
+    source_type: Literal[SystemLogSourceType.K8S_EVENT] = SystemLogSourceType.K8S_EVENT
+
+
+class RayHeadNodeEventLog(pydantic.BaseModel):
+    """A system-level log line produced by a Ray head node event."""
+
+    tag: str
+
+    log: str
+
+    source_type: Literal[
+        SystemLogSourceType.RAY_HEAD_NODE
+    ] = SystemLogSourceType.RAY_HEAD_NODE
+
+
+class RayWorkerNodeEventLog(pydantic.BaseModel):
+    """A system-level log line produced by a Ray head node event."""
+
+    tag: str
+
+    log: str
+
+    source_type: Literal[
+        SystemLogSourceType.RAY_WORKER_NODE
+    ] = SystemLogSourceType.RAY_WORKER_NODE
+
+
+SysLog = Annotated[
+    Union[K8sEventLog, RayHeadNodeEventLog, RayWorkerNodeEventLog],
+    pydantic.Field(discriminator="source_type"),
+]
+
+
+class SysMessage(NamedTuple):
+    """
+    A pair of ``[timestamp, syslog]``.
+
+    Based on:
+    https://github.com/zapatacomputing/workflow-driver/blob/972aaa3ca75780a52d01872bc294be419a761209/openapi/src/resources/workflow-run-logs.yaml#L18
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** ‏This should be the system logs defintion.

Reply:
It should indeed, updated :)